lib/shelter.rb

```diff
@@ -1,2 +1,20 @@
 class Shelter
+
+  attr_reader :name, :capacity, :pets
+
+  def initialize(name,capacity,pets=[])
+    @name = name
+    @capacity = capacity
+    @pets = pets
+  end
+
+
+  def add_pet(pet)
+    @pets.push(pet)
+  end
+
+  def call_pets(pet)
+    @call_pets = @pets.collect{|pet| pet.append("#{pet}!")}
```

Comment:
`collect` is the correct enumerable to use here.  But, `append` is not a string method (it can only be called on an array).  What you want to collect is the string that you have in the append argument.

```
@pets.collect { |pet| "#{pet}!" }
```

Also, you don't _need_ to assign the return value of `collect` to a variable in this case.  You would only need to do that if you need to use that new array for another operation within the same method.